tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
+
+		if ((x[n] < min || x[n] > max) || (y_check < fTofdTotLow || y_check > fTofdTotHigh))
+		{
+			delete histo_py;
+			continue;
+		}
+		if(x[n] == 0.) cout<<"x=0 ";
+		if (histo_py->GetMaximum() > coarse_threshold)
+		{
+			Double_t fitrange = 5.;
+			TF1* fgaus = new TF1("fgausX", "gaus(0)", /*histo_py->GetXaxis()->GetBinCenter(binmax)*/y_check - fitrange, /*histo_py->GetXaxis()->GetBinCenter(binmax)*/y_check + fitrange);
+			fgaus->SetParameter(1, y_check);
+			fgaus->SetParLimits(1,y_check - (fitrange*1.25),y_check + (fitrange*1.25));
+			histo_py->Fit(fgaus, "QR");
+			Double_t fix = fgaus->GetParameter(1);
+			//y[n] = y_check;
+			y[n] = fix;
+			if(x[n] == 0.) cout<<"y "<<y_check<<" "<<y[n]<<endl;
+			fgaus->Draw();
+			cout<<"y "<<y_check<<" y_new "<<fix<<endl;
+			n++;
+			delete l;
+			delete fgaus;
+		}
+		delete histo_py;
+	}
+	for(int c = 0; c<=n;c++){
+		cout<<"n "<<c<<" x "<<x[c]<<" y "<<y[c]<<endl;
+	}
+	gr1 = new TGraph(n, x, y);
+	gr1->SetTitle("Points found for fitting; x position in cm; sqrt(tot1*tot2)");
+	gr1->Draw("A*");
+	std::cout << "Start fitting\n";
+	TF1* f1 = new TF1("f1", "pol3", min, max);
+	f1->SetLineColor(2);
+	gr1->Fit("f1", "Q", "", min, max);
+	for (Int_t j = 0; j <= 3; j++)
+	{
+		para[j] = f1->GetParameter(j);
+		std::cout << "Parameter: " << para[j] << "\n";
+	}
+	// fit again but with more information and better cuts
+	std::cout << "Fit again with more information\n";
+	n = 0;
+	cfit_smiley->cd(4);
+	for (Int_t i = 1; i < histo2->GetNbinsX(); i++)
+	{
+		Double_t pos = histo2->GetXaxis()->GetBinCenter(i);
+		Double_t ymean = f1->Eval(pos);
+		histo2->SetAxisRange(ymean - 5., ymean + 5., "Y");
+		histo2->Draw("colz");
+		TH1F* histo_py = (TH1F*)histo2->ProjectionY("histo_py", i, i, "");
+		//histo_py->Rebin(rebin2);
+		histo_py->Draw();
+		x[n] = histo2->GetXaxis()->GetBinCenter(i);
+		Int_t binmax = histo_py->GetMaximumBin();
+		Double_t y_check = histo_py->GetXaxis()->GetBinCenter(binmax);
+		//y[n] = histo_py->GetXaxis()->GetBinCenter(binmax);
+
+		if ((x[n] < min || x[n] > max) || (y_check < fTofdTotLow || y_check > fTofdTotHigh))
+		{
+			delete histo_py;
+			continue;
+		}
+		if (histo_py->GetMaximum() > fine_threshold){
+			Double_t fitrange = 5.;
+			TF1* fgaus2 = new TF1("fgaus2", "gaus(0)", y_check - (fitrange), y_check + (fitrange));
+			fgaus2->SetParameter(1, y_check);
+			fgaus2->SetParLimits(1,y_check - (fitrange*1.25),y_check + (fitrange*1.25));
+			histo_py->Fit(fgaus2, "QR");
+			y[n] = fgaus2->GetParameter(1);
+			//y[n] = y_check;
+			cout<<"fine y "<<y_check<<" y_new "<<y[n]<<endl;
+			fgaus2->Draw();
+			n++;
+		}
+		delete histo_py;
+	}
+	for(int c = 0; c<=n;c++){
+		cout<<"n_fine "<<c<<" x "<<x[c]<<" y "<<y[c]<<endl;
+	}
+	gr2 = new TGraph(n, x, y);
+	gr2->SetTitle("More information;x position in cm;sqrt(q1*q2)");
+	gr2->Draw("A*");
+	f1->DrawCopy("SAME");
+	TF1* f2 = new TF1("f2", "pol3", min, max);
+	f2->SetParameters(para[0], para[1], para[2], para[3]);
```

Comment:
Now I'm still unsure why you changed the charge calibration from division to multiplication of this function.
But maybe I missed something in your code.

Reply:
Enis looked into this. When he flipped the sign, the data converged. Before that, it diverged

Reply:
Interesting. Is there a mixup somewhere that causes this issue? I would expected something like I described in https://github.com/R3BRootGroup/R3BRoot/pull/842#discussion_r1164626320